jupyterhub/templates/hub/deployment.yaml

```diff
@@ -44,7 +44,7 @@ spec:
           - jupyterhub
           - --config
           - /srv/jupyterhub_config.py
-          {{ if eq .Values.hub.db.type "sqlite-pvc" }}
```

Comment:
As this is, users *cannot* disable upgrade for sqlite-pvc. Do we want them to? If so, a null/'auto' default might be more appropriate so we can preserve opt-in for others and opt-out for sqlite-pvc

Reply:
Hmm, go templates is a shitty mixture of 'static typing' and 'false, nil, 0, "" are all false'. I'm trying to figure out how to differentiate between nil and false. 'auto' as a string runs into the static typing problem, where the template var is string some of the time but a bool some of the time...

Reply:
Done! lmk what you think of this :)

Reply:
oof. That sure is not as nice as `is None`, but I think the logic is correct, and it gives the option, which is good.